### ISSUE TITLE
Document PKCS5_pbe2_set_iv and sibling(s) as well as PBKDF2 procedure for PKCS8

### ISSUE THREAD

**Issue by clauderobi** [openssl/openssl#12519]
I am using openSSL 1.1.1g. 

In order to be compatible with Web Crypto API (and its wrapper openCrypto) regarding encryption RSA keys, The pKCS8 (and PBKDF2) is needed. The documentation is lacking info on PKCS5_pbe2_set_iv. Is it possible to create it? Thanks.

**Comment by t8m** [comment]
Please note we are looking for community help implementing this issue. Please comment if you are interested in implementing this feature or if you have reasons why this feature is important enough so we should consider implementing it with higher priority than other requested features. The issue will be closed after 3 months otherwise.

**Comment by zoeychappell** [comment]
Hello!

I'm interested in implementing this issue.

@bbbrumley

**Comment by bbbrumley** [comment]
@zoeychappell seems like it's worth looking into. @t8m what do you think?

**Comment by kovan** [comment]
This appears to have been addressed. PKCS5_pbe2_set_iv() and related functions are documented in doc/man3/PKCS5_PBE_keyivgen.pod. The documentation includes the function signatures in SYNOPSIS and descriptions in the DESCRIPTION section.

**Comment by bbbrumley** [comment]
@kovan I did notice it's in the man page, but my impression was, the description there is still lacking? @t8m what do you think?

**Comment by zoeychappell** [comment]
I believe that it would be valuable to add clarification to the DESCRIPTION section of doc/man3/PKCS5_PBE_keyivgen.pod. My suggestions:

1. Adding a brief note that PKCS5_pbe2_set_iv() (and related functions) encode the key derivation parameters (salt, iteration count, PRF) and cipher parameters (such as the IV).
2. Include a short description of the parameters that aren't currently explained, for example, _aiv_.  
3. Clarify what "associated parameters" are in the Algorithm Identifier Creation section. 

Let me know what you think! 

@bbbrumley

**Comment by bbbrumley** [comment]
@zoeychappell I agree with your assessment, and your plan makes sense! I suggest moving forward with your plan, please link this issue when you file your PR
